chips/imxrt1050/src/ccm.rs

```diff
@@ -0,0 +1,711 @@
+use kernel::common::registers::{register_bitfields, ReadOnly, ReadWrite};
+use kernel::common::StaticRef;
+use kernel::ClockInterface;
+
+// Clock Controller Module
+#[repr(C)]
+struct CcmRegisters {
+    // CCM control register
+    ccr: ReadWrite<u32, CCR::Register>,
+    _reserved1: [u8; 4],
+    // CCM status register
+    csr: ReadOnly<u32, CSR::Register>,
+    // CCM Clock Switcher Register
+    ccsr: ReadWrite<u32, CCSR::Register>,
+    // unimplemented
+    _reserved2: [u8; 20],
+    cscdr1: ReadWrite<u32, CSCDR1::Register>,
+    _reserved3: [u8; 44],
+    clpcr: ReadWrite<u32, CLPCR::Register>,
+    _reserved4: [u8; 16],
+    // clock gating register 0
+    ccgr0: ReadWrite<u32, CCGR0::Register>,
+    // clock gating register 1
+    ccgr1: ReadWrite<u32, CCGR1::Register>,
+    // clock gating register 2
+    ccgr2: ReadWrite<u32, CCGR2::Register>,
+    // clock gating register 3
+    ccgr3: ReadWrite<u32, CCGR3::Register>,
+    // clock gating register 4
+    ccgr4: ReadWrite<u32, CCGR4::Register>,
+    // clock gating register 5
+    ccgr5: ReadWrite<u32, CCGR5::Register>,
+    _reserved6: [u8; 8],
+}
+
+register_bitfields![u32,
+    CCR [
+        /// Enable for REG_BYPASS_COUNTER
+        RBC_EN OFFSET(27) NUMBITS(1) [],
+
```

Comment:
Nitpick, but probably should be formatted without additional newlines as most register definitions around Tock?

Reply:
I fixed the issues and created a new pull request, since this branch was not up to date with the master. Thank you for your input! I will close this pull request and we can continue on the new one.